src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -359,20 +360,26 @@ public Visitable visit(MethodDeclaration method, Void p) {
   public Visitable visit(Parameter para, Void p) {
     if (insideTargetMethod) {
       Type type = para.getType();
+      // an unknown type plays the role of a null object for lambda parameters that have no explicit
+      // type declared. However, we also want to avoid trying to solve declared lambda params (it
+      // will
+      // fail, despite the fact that the code should be compilable).
+      boolean isLambdaParam = type.isUnknownType() || isLambdaParam(para);
       if (type.isUnionType()) {
         resolveUnionType(type.asUnionType());
-      }
-      // an unknown type plays the role of a null object for lambda parameters that have no explicit
-      // type declared
-      else if (!type.isUnknownType()) {
+      } else if (!isLambdaParam) {
         // Parameter resolution (para.resolve()) does not work in catch clause.
         // However, resolution works on the type of the parameter.
         // Bug report: https://github.com/javaparser/javaparser/issues/4240
         ResolvedType paramType;
         if (para.getParentNode().isPresent() && para.getParentNode().get() instanceof CatchClause) {
           paramType = para.getType().resolve();
         } else {
-          paramType = para.resolve().getType();
+          try {
+            paramType = para.resolve().getType();
+          } catch (UnsupportedOperationException e) {
+            throw new RuntimeException("cannot solve: " + para, e);
```

Comment:
I added this as a debugging aid, but I don't see a reason to remove it - it's useful to know what causes crashes